service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -169,6 +169,13 @@ public Workspace getWorkspace(UUID id, AuthenticatedUserRequest userRequest) {
     return validateWorkspaceAndAction(userRequest, id, SamConstants.SamWorkspaceAction.READ);
   }
 
+  /** Retrieves an existing workspace by userFacingId */
+  @Traced
+  public Workspace getWorkspaceByUserFacingId(String userFacingId, AuthenticatedUserRequest userRequest) {
```

Comment:
I would prefer to do this operation in one DB transaction instead of two.
You would put the Sam authz call here instead of the `validateWorkspaceAndAction`

If we get a second userFacingId-based API, we will probably want a `validateWorkspaceAndActionByUFID` method.

Reply:
Done